pkg/controller/lambda/function/setup.go

```diff
@@ -271,18 +272,48 @@ type updater struct {
 	client svcsdkapi.LambdaAPI
 }
 
+func (u *updater) isLastUpdateStatusSuccessful(cr *svcapitypes.Function) error {
+	for {
+		out, err := u.client.GetFunction(&svcsdk.GetFunctionInput{
+			FunctionName: aws.String(meta.GetExternalName(cr)),
+		})
+		if err != nil {
+			return err
+		}
+		if aws.StringValue(out.Configuration.LastUpdateStatus) == svcsdk.LastUpdateStatusSuccessful {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
Is 1-second sleep ok here? Should I use something different?

Reply:
@negz are we allowed to sleep here holding the goroutine? Should we instead be able to request a later via a return value in `managed.ExternalUpdate`? That might not be trivial as I guess you might want exponential backoff...

Reply:
Doesn't return with `managed.ExternalUpdate` start whole `update` from the beginning? The issue is, we MUST check for status between `UpdateFunctionCodeWithContext` and `UpdateFunctionConfigurationWithContext`.

Reply:
Ideally, on the next update, we will not call `UpdateFunctionCodeWithContext` if we return early (on a timeout waiting for isLastUpdateStatusSuccessful), because we should only call that code conditionally when the function code needs updating, right?

Anyway, if we should have a loop here, I think there should be a limit here, do we have any idea about the worst and normal case here? I think 5-15s is a good choice, especially if it typically takes less than a second.

_Maybe_ it would be better wth a backoff instead of a fixed delay, perhaps using: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff

And as mentioned initially, if possible, only call `UpdateFunctionCodeWithContext` if needed. This way, we should handle both a situation where the update completes quickly, and also a case where it requires an extra reconcile due to high latency in AWS?

Reply:
I think the limit is 1m, I noticed this value:
```
src/crossplane-runtime (metrics=)$ rg reconcileTimeout
pkg/reconciler/managed/reconciler.go
44:	reconcileTimeout     = 1 * time.Minute
655:		timeout:             reconcileTimeout,
```

So maybe a limit around 15s - 55s would be a good start? I think we can iterate on the other changes (if needed at all) in later improvements, as I understand it this is an important issue to get fixed now.

Reply:
So, should I go for static limit? Exponential? Will you be able to approve and merge this?

Reply:
I think it makes more sense to return here without an error if the status is non-successful and check again during the next update.